ethereum/geth_client.go

```diff
@@ -347,10 +347,12 @@ func (gc *gethClient) SubmitTransactionWithRetries(contractMethod interface{}, o
 		}
 
 		if err == nil {
+			log.Infof("TX %s successfully sent with nonce %s", tx.Hash().String(), opts.Nonce.String())
 			return tx, nil
 		}
 
-		if (err.Error() == ErrTransactionUnderpriced.Error()) || (err.Error() == ErrNonceTooLow.Error()) {
+		if strings.Contains(err.Error(), ErrTransactionUnderpriced.Error()) || strings.Contains(err.Error(), ErrNonceTooLow.Error()) ||
```

Comment:
Fixing this will not ensure that all txs will always go through, since they can still fail silently in the processing eth node.